include/swift/SIL/SILFunction.h

```diff
@@ -1234,6 +1234,29 @@ class SILFunction
     return V && V->getAttrs().hasAttribute<AlwaysEmitIntoClientAttr>();
   }
 
+  /// Returns true if this function belongs to a declaration that
+  /// has `@_used` attribute.
+  bool markedAsUsed() const {
+    if (!hasLocation())
+      return false;
+
+    auto *V = getLocation().getAsASTNode<ValueDecl>();
+    return V && V->getAttrs().hasAttribute<UsedAttr>();
+  }
+
+  /// Returns a SectionAttr if this function belongs to a declaration that
+  /// has `@_section` attribute.
+  SectionAttr *getSectionAttr() const {
```

Comment:
We need to think about inlining behavior. What should happen if a function with a section attribute is inlined into a function without (or a different) section attribute? Should inlining be prevented in this case?

Reply:
Good point. My sense is that we should mirror what `__attribute__((section))` does in Clang: It does not prevent inlining or interact with it in any way. What the attribute does is that it makes sure that the standalone/top-level function is emitted into that section, but other functions are still free to inline the code from it (as long as we *also* emit the standalone function that is in the right section). If the user wants to prevent inlining, they can use another attribute for that, `__attribute__((noinline))` / `@inline(never)`.

WDYT?